core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -65,7 +65,7 @@ sealed abstract class Parser[+A] {
     *
     * To require the entire input to be consumed, see `parseAll`.
     */
-  final def parse(str: String): Either[Parser.Error, (String, A)] = {
+  final def parse0(str: String): Either[Parser.Error, (String, A)] = {
```

Comment:
can we not change the names of these methods?

Reply:
This is the "taking it to the extreme" part I mean in the main comment. I'm not married to the idea. The thought was, there are very few grammars that accept the empty string. Not calling a combinator with them but actually using them to call parse on could have an "alerting" name on them. Coincidentally, it did lead to finding Parsers being widened to Parser1 where they didn't need to.

The ugly parse0/parseAll0 show up here only in testing the subgrammar for white space, which is exactly the sore thumb effect it had in mind: careful, this might consume no input.

But like I said, I'm not married to the idea, and happy to take that commit out.